File: src/main/java/art/chibi/telemetry/TelemetryComponent.java
package art.chibi.telemetry;

import dagger.BindsInstance;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.logging.Logger;

@dagger.Component(modules = TelemetryModule.class)
public interface TelemetryComponent {

    void inject(TelemetryPlugin plugin);

    @dagger.Component.Builder
    interface Builder {

        @BindsInstance
        Builder pluginInstance(JavaPlugin plugin);

        @BindsInstance
        Builder logger(Logger logger);

        @BindsInstance
        Builder telemetryConfig(Config config);

        TelemetryComponent build();
    }
}

File: src/main/java/art/chibi/telemetry/di/TelemetryComponent.kt
package art.chibi.telemetry.di

import art.chibi.telemetry.TelemetryPlugin
import art.chibi.telemetry.config.TelemetryConfig
import dagger.BindsInstance
import dagger.Component
import org.bukkit.plugin.java.JavaPlugin
import java.util.logging.Logger

@Component(modules = [TelemetryModule::class])
interface TelemetryComponent {

    fun inject(plugin: TelemetryPlugin)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun pluginInstance(plugin: JavaPlugin): Builder

        @BindsInstance
        fun logger(logger: Logger): Builder

        @BindsInstance
        fun telemetryConfig(config: TelemetryConfig): Builder

        fun build(): TelemetryComponent
    }
}